sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/options/ShareFileRenameOptions.java

```
@@ -170,4 +172,24 @@ public ShareFileRenameOptions setMetadata(Map<String, String> metadata) {
         this.metadata = metadata;
         return this;
     }
+
+    /**
+     * Gets the {@link ShareFileHttpHeaders}. Currently, only content type is respected. Others are ignored.
+     *
+     * @return The {@link ShareFileHttpHeaders}.
+     */
+    public ShareFileHttpHeaders getHeaders() {
+        return this.headers;
+    }
+
+    /**
+     * Sets the {@link ShareFileHttpHeaders}.  Currently, only content type is respected. Others are ignored.
```

Would service throw at not-respected headers? How would customer know they're passing other headers by mistake?

---

We don't even put the non respected headers on the wire, so they're silently ignored. I believe we have other instances of this in the storage sdk and we haven't chosen to throw on the client side because the service may accept them at some point

---

That said, I am fine to throw if we decide that's better

---

Not putting them on the wire is different from passing them to service and letting service take responsibility of handling them (should they be ignored or cause errors).
There was a similar problem with request conditions , see https://github.com/Azure/azure-sdk-for-net/issues/19770 and https://github.com/Azure/azure-sdk-for-net/issues/21977 .

---

So it looks like the preference is to validate and throw?

---

Yes if we want to stick with this type. Other options are to create new type for this or flatten "Content Type" into `ShareFileRenameOptions` until more headers are actually supported. Each choice is a trade-off of some sort.